src/mbgl/renderer/pattern_atlas.cpp

```diff
@@ -0,0 +1,109 @@
+#include <mbgl/renderer/pattern_atlas.hpp>
+#include <mbgl/gfx/upload_pass.hpp>
+#include <mbgl/gfx/context.hpp>
+
+namespace mbgl {
+
+namespace {
+
+// When copied into the atlas texture, image data is padded by one pixel on each side. Icon
+// images are padded with fully transparent pixels, while pattern images are padded with a
```

Comment:
> Icon images are padded
I guess it's good to mention that icon images are not added here but through ImageManager.